### PR TITLE
New Rules Engine and Network plugin

### PR DESCRIPTION
**Pull request by muenchris**

### CODE REVIEW

File: src/093 - CDMyNetwork/C-DMyNetwork/packages.config [hunk 1]
```diff
@@ -0,0 +1,5 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (MarkusHorstmann) [inline src/093 - CDMyNetwork/C-DMyNetwork/packages.config]: We can remove this file

File: src/093 - CDMyNetwork/C-DMyNetwork/C-DMyNetwork.csproj [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ -->
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;net45;net40;net35</TargetFrameworks>
+        <Version>4.302.0</Version>
+        <Authors>Chris Muench</Authors>
+        <Company>C-Labs</Company>
+        <Product>C-DMyNetwork</Product>
+        <Description>Network Services</Description>
+        <Copyright>(C) 2017-2020 C-Labs</Copyright>
+        <ProjectName>$(AssemblyName).$(TargetFramework)</ProjectName>
```
Comment (MarkusHorstmann) [inline src/093 - CDMyNetwork/C-DMyNetwork/C-DMyNetwork.csproj:14]: Should generate a nuget package for this

File: src/093 - CDMyNetwork/C-DMyNetwork/cdeMyNetwork.cs [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    }
+
+    partial class MyNetworkServices : ThePluginBase
+    {
+        #region ICDEPlugin
+        /// <summary>
+        /// This constructor is called by The C-DEngine during initialization in order to register this service
+        /// </summary>
+        /// <param name="pBase">The C-DEngine is creating a Host for this engine and hands it to the Plugin Service</param>
+        public override void InitEngineAssets(IBaseEngine pBase)
```
Comment (MarkusHorstmann) [inline src/093 - CDMyNetwork/C-DMyNetwork/cdeMyNetwork.cs:30]: Should we use attributes here?

File: src/093 - CDMyNetwork/C-DMyNetwork/C-DMyNetwork.csproj (hunk 2 above)
Comment (muenchris) [reply src/093 - CDMyNetwork/C-DMyNetwork/C-DMyNetwork.csproj:14]: what is missing?

File: src/149 - CDMyRulesEngine/CDMyRulesEngine/CDMyRulesEngine.csproj [hunk 4]
```diff
@@ -0,0 +1,67 @@
+﻿<Project Sdk="Microsoft.NET.Sdk" >
+  <!--
+ SPDX-FileCopyrightText: 2009-2020 TRUMPF Laser GmbH, authors: C-Labs
+ SPDX-License-Identifier: MPL-2.0
+ -->
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;net45;net40;net35</TargetFrameworks>
```
Comment (MarkusHorstmann) [inline src/149 - CDMyRulesEngine/CDMyRulesEngine/CDMyRulesEngine.csproj:7]: Should we remove net40 target then?

File: src/093 - CDMyNetwork/C-DMyNetwork/cdeMyNetwork.cs (hunk 3 above)
Comment (muenchris) [reply src/093 - CDMyNetwork/C-DMyNetwork/cdeMyNetwork.cs:30]: next update...when the telnet device type is coming back after cdeTelnet shared project is ready with nuget

File: src/149 - CDMyRulesEngine/CDMyRulesEngine/CDMyRulesEngine.csproj (hunk 4 above)
Comment (muenchris) [reply src/149 - CDMyRulesEngine/CDMyRulesEngine/CDMyRulesEngine.csproj:7]: if we remove 40 then also remove 3.5 either both or none

File: src/093 - CDMyNetwork/C-DMyNetwork/C-DMyNetwork.csproj (hunk 2 above)
Comment (MarkusHorstmann) [reply src/093 - CDMyNetwork/C-DMyNetwork/C-DMyNetwork.csproj:14]: <GeneratePackageOnBuild>true</GeneratePackageOnBuild>

File: src/149 - CDMyRulesEngine/CDMyRulesEngine/CDMyRulesEngine.csproj (hunk 4 above)
Comment (MarkusHorstmann) [reply src/149 - CDMyRulesEngine/CDMyRulesEngine/CDMyRulesEngine.csproj:7]: If it doesn't work why compile for it? (referring to the comment below, BTW).

Comment (MarkusHorstmann) [reply src/149 - CDMyRulesEngine/CDMyRulesEngine/CDMyRulesEngine.csproj:7]: The comment was outdated

**Review by MarkusHorstmann — APPROVED** [review]

**Pull request merged**